csrc/welford.cu

```diff
@@ -291,8 +292,8 @@ __global__ void welford_kernel(
     }
   }
 
-  static __shared__ int s_mem[160];
-  accscalar_t* s_mem_ac = (accscalar_t*) &s_mem[32];
+  static __shared__ int s_mem[MAX_BLOCK_SIZE/WARP_SIZE];
+  static __shared__ accscalar_t s_mem_ac[MAX_BLOCK_SIZE/WARP_SIZE*2];
```

Comment:
Why is smem size reduced from 160 to MAX_BLOCK_SIZE/WARP_SIZE = 8 and s_mem_ac size reduced from 32 to  MAX_BLOCK_SIZE/WARP_SIZE*2 = 16? It seems like s_mem is used for storing the value from each lane (count[lane]). Should not it be of size at least 64?  s_mem_ac seems to be used for storing the value from each lane too but the offset is different (x[lane*2+1]). I think it should be of size at least 128. Did we test this kernel?

Reply:
Good catch.  I will update.

Reply:
Great catch, thank you.

Reply:
Fixed in latest commit.